cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/SparkProgramStatusMetricsProvider.java

```diff
@@ -76,6 +80,12 @@ public SparkProgramStatusMetricsProvider(CConfiguration cConf, MetricsCollection
     this.maxTerminationMinutes = cConf.getInt(Constants.Spark.SPARK_METRICS_PROVIDER_MAX_TERMINATION_MINUTES, 5);
     this.cConf = cConf;
     this.metricsCollectionService = metricsCollectionService;
+    this.retryStrategy = RetryStrategies.fromConfiguration(this.cConf,
+                                                           Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX);
+    this.sparkHistoricBaseURL = cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_HOST);
+    if (StringUtils.isEmpty(sparkHistoricBaseURL)) {
```

Comment:
Generally we prefer the guava library instead of the apache commons (ideally we want to minimize dep). You can use `Strings.isNullOrEmpty()` from guava.Generally we prefer the guava library instead of the apache commons (ideally we want to minimize dep). You can use `Strings.isNullOrEmpty()` from guava.